lessons/intro/flask/index.md

```diff
@@ -373,7 +377,110 @@ V šabloně pak například:
+HTTP odpovědí na dotazy (ať už ve formě webové stránky, JSON odpovědi či jiném
+formatu), ale také vypisovat různé chybové, informační či ladící hlášky na
+straně serveru. Možností je použít například `print`, ale s ním pravděpodobně
+brzy narazíte na problém, že jednak výstup není konzistentní s jinýmy hláškami
+produkovanými Flaskem, ale především, že pro různé typy výpisů, časová razítka,
+přesměrování logu do souboru a další potřebujete vytvářet spoustu logiky kolem
+namísto vytváření samotné webové aplikace.
+
+Řešením je použít standardní logovací modul [logging], který řeší vše potřebné 
+(úrovně zpráv, filtry, formátování časového razítka a dalších meta-informací o
+běhu programu) a výstup bude konzistentní s jinými aplikacemi (jiní správci 
+vaší webové aplikace pak nebudou z formátu výstupů zmatení). Protože používáme
+Flask a ten také [loguje tímto modulem](https://flask.palletsprojects.com/en/1.1.x/logging/),
+stačí použít předpřipravený `app.logger`.
+
+```python
+from flask import Flask
+
+app = Flask(__name__)
+app.logger.debug('I\'ve just initialized my Flask app')
+
+@app.route('/')
+def index():
+    app.logger.warning('Someone is accessing index page!')
+    return 'Index Page'
+```
+
+Ve výchozím nastavení se loguje pouze od úrovně upozornění výše (`warning`, 
+`error`, `critical`). Při spuštění aplikace v ladícím režimu se loguje vše
+(navíc i `debug` a `info`). Aktuální úroveň je možné také změnit pomocí metody
+`setLevel`, viz dokumentace modulu [logging].
+
+[logging]: (https://docs.python.org/3/library/logging.html#module-logging)
+
+### Větší Flask aplikace
+
+Flask je sice označován jako mikroframework, to ale neznamená, že jej nelze 
+použít na větší a složitější webové aplikace. Samozřejmě stojí opět za zvážení, 
+zda zvolit Flask nebo například Django.
+
+#### create_app factory
+
+Mimo vytváření Flask aplikace přímo v modulu, je možné aplikaci tvořit pomocí
+funkce, tzv. [*application_factory*](https://flask.palletsprojects.com/en/1.1.x/patterns/appfactories/), 
+standardně pojmenované `create_app`. Takový přístup má výhodu, že aplikace se 
+neinicializuje hned při importu modulu, ale až při zavolání funkce. Voláním funkce
+můžete navíc předat i konfigurační parametry (typicky cesta ke konfiguračnímu 
+souboru). Díky tomu lze snadněji vytvářet Flask aplikace s různými konfiguracemi
+pro testování nebo dokonce vytvářet více Flask aplikací v rámci jednoho Python skriptu.
+
+```python
+def create_app(config=None):
+    app = Flask(__name__)
+
+    app.config.from_pyfile(config or 'config.py')
+    app.config['the_answer'] = 42
+    app.secret_key = os.environ.get('MY_SECRET', None)
+
+    return app
+```
+
+#### Blueprint moduly
+
+Ve velkých webových aplikacích je již vhodné seskupovat jednotlivé pohledy do
+samostatných celků. K tomuto účelu slouží ve Flasku [blueprinty] (hezky česky 
+„modrák” nebo „modrotisk”). Výhodou je, že můžete vytvořit blueprint (instanci
```

Comment:
modrák? máš na to nějakou referenci?

Reply:
https://cs.wikipedia.org/wiki/Diazotypie 😄 můžu to tam dát jako odkaz...

Reply:
asi dej, slyším to poprvé